src/neuroconv/basedatainterface.py

```diff
@@ -1,28 +1,64 @@
 """Authors: Cody Baker and Ben Dichter."""
 from abc import abstractmethod, ABC
 import uuid
-from typing import Optional
+from typing import Optional, Dict, Any, List
 
 from pynwb import NWBFile
+from pydantic.schema import model_schema
 
-from .utils import get_base_schema, get_schema_from_method_signature
+from .utils import (
+    get_base_schema,
+    get_schema_from_method_signature,
+    get_pydantic_model_from_method_signature,
+    FilePathType,
+)
 
 
 class BaseDataInterface(ABC):
-    """Abstract class defining the structure of all DataInterfaces."""
+    """
+    Abstract class defining the structure of all DataInterfaces.
+
+    Attributes
+    ----------
+    source_data : dict
+        Structure of input arguments used in upstream initialization.
+    source_data_to_validate : dict, default: None
```

Comment:
```suggestion
    source_data_to_validate : dict, optional
```

This is the correct way to annotate when the default is None